ms/MeasurementSets/MSIter.h

```diff
@@ -60,11 +60,11 @@ class MSInterval : public BaseCompare
   explicit MSInterval(Double interval) : interval_p(interval), offset_p(0) {}
     virtual ~MSInterval() {}
     virtual int comp(const void * obj1, const void * obj2) const;
-    Double getOffset() {return offset_p;}
-    void setOffset(Double offset) {offset_p=offset;}
-    Double getInterval() {return interval_p;}
+    Double getOffset() const {return offset_p;}
+    virtual void setOffset(Double offset) {offset_p=offset;}
+    Double getInterval() const {return interval_p;}
     void setInterval(Double interval) {interval_p=interval;}
-private:
+protected:

```

Comment:
Having instance variable as protected is generally not a good practice.  Unnecessary as their are getters and setters available.